### PR TITLE
Section 6 Comments

### LINKED ISSUE #244: Section 6.4.1
**Organization / Program:** Slandala <br>**Section:** 6.4.1 <br>**PDF Page:** None<br>**PDF Line(s):** 1674<br>**Comment:** Are there requrieemnts to secure or cahnge passwords, PINs or other activation data?<br><br>**Suggested Change:** None

### LINKED ISSUE #245: Section 6.6.1
**Organization / Program:** Slandala <br>**Section:** 6.6.1 <br>**PDF Page:** None<br>**PDF Line(s):** 1723<br>**Comment:** Do offline CAs require AV protection?<br><br>**Suggested Change:** None

### LINKED ISSUE #259: Section 6.1.1.1
**Organization / Program:** Scott S. Perry CPA, PLLC <br>**Section:** 6.1.1.1 <br>**PDF Page:** 60<br>**PDF Line(s):** 1525<br>**Comment:** Grammar is askew suggesting that the Qualified Auditor erroneously does the recording.<br><br>**Suggested Change:** Rewrite to state: "have a Qualified Auditor witness the CA Key Pair generation process or review a recording of a video"

### LINKED ISSUE #260: Section 6.1.3
**Organization / Program:** Scott S. Perry CPA, PLLC <br>**Section:** 6.1.3 <br>**PDF Page:** 61<br>**PDF Line(s):** 1559<br>**Comment:** The section lacks the requirement that delivery of public key must be to an Applicant/Subscriber that possesses the associated Private Key.<br><br>**Suggested Change:** ADD: The certificate request process shall ensure that the Applicant possesses the Private Key associated with the Public Key presented for certification.

### LINKED ISSUE #261: Section 6.2.2
**Organization / Program:** Scott S. Perry CPA, PLLC <br>**Section:** 6.2.2 <br>**PDF Page:** 63<br>**PDF Line(s):** 1612<br>**Comment:** CP does not state in this section (and section 5.2.4) that the Auditor (both Audit Administrator and Qualified Auditor) shall be excluded when establishing two person control.<br><br>**Suggested Change:** CP should state in this section (and section 5.2.4) that the Auditor (both Audit Administrator and Qualified Auditor) shall be excluded when establishing two person control.

### LINKED ISSUE #262: Section 6.2.4
**Organization / Program:** Scott S. Perry CPA, PLLC <br>**Section:** 6.2.4 <br>**PDF Page:** 63<br>**PDF Line(s):** 1624<br>**Comment:** CP is silent on the level of protection that is required for storage of the private signature key off site.<br><br>**Suggested Change:** CP should specify that the private signature key shall be stored off-site in a safe, secure storage facility separate from the CA.

### LINKED ISSUE #263: Section 6.6.1
**Organization / Program:** Scott S. Perry CPA, PLLC <br>**Section:** 6.6.1 <br>**PDF Page:** 66<br>**PDF Line(s):** 1714<br>**Comment:** The introduction line refers to system development controls of the RA but does not mention it in the underlying bullets.  The re is no mention of the CSA (CRL, OCSP) considered a critical component of the PKI<br><br>**Suggested Change:** Expand and consistently include the CA, RA and CSA in the scope of system development controls.

### LINKED ISSUE #296: Section 6.1.4
**Organization / Program:** FPKIMA <br>**Section:** 6.1.4 <br>**PDF Page:** 61<br>**PDF Line(s):** 1563<br>**Comment:** Re-key is not authorized per this CP, nor is certificate renewal.  So what does this section mean when it states "When a Subordinate CA updates its signature key pair…"<br><br>**Suggested Change:** Clarify this section to explain what Subordinate CA updates refer to.  Is this a new subordinate CA that replaces the legacy CA?

### LINKED ISSUE #297: Section 6.3.2
**Organization / Program:** FPKIMA <br>**Section:** 6.3.2 <br>**PDF Page:** 65<br>**PDF Line(s):** 1667-1668<br>**Comment:** Line 813-814 states:

The CA SHALL verify that the OCSP Delegated Responder certificate expiration date SHALL NOT exceed 825 days from the date of the initial certificate issuance.

Line 1667-1668 states:

Subscriber Certificates issued for delegated OCSP responders SHALL have a Validity Period no greater than 45 days.

This seems inconsistent.  Should Delegated OCSP Responder certificates be valid for 45 days, or can they be valid for up to 825 days?<br><br>**Suggested Change:** Update one section to provide consistency.

### LINKED ISSUE #298: Section 6.5.1
**Organization / Program:** FPKIMA <br>**Section:** 6.5.1 <br>**PDF Page:** 65<br>**PDF Line(s):** 1682-1686<br>**Comment:** Although multifactor would be a great thing to have, most vendors do not fully support multifactor authentication.  For example, no U.S. based FIPS 140-2 Level 3 HSMs that support 2 factor authentication to the OS. Two factor authentication for those only occurs when accessing the partitions/security worlds that store the keys using the PED. 

There is a similar issue for other operating systems.  For example, smart card authentication to a RHEL or Microsoft OS would require being connected to a network and/or domain.  Other options, such as OTP, still require network connectivity. Additionally, what are the "component systems"?  Many vendors also do not support smart card authentication for stand alone OCSP systems.  Connecting an offline system to a network just for the purpose of two factor authentication unnecessarily increases the attack surface when compared to a dedicated offline system that is never connected to a network. 

“All CAs SHALL implement multifactor authentication for all Trusted Role accounts capable of directly causing certificate issuance or authenticating to Certificate Systems. All Trusted Roles SHALL use a unique credential created by or assigned to a single person for identification and authentication.

All CAs SHALL implement multifactor authentication for all access to component systems including operating system and software.”
<br><br>**Suggested Change:** Update to align with latest network security requirements of "multifactor or multi-party". Define what the "component systems" include, and include the term "where possible."  Removing the words "or authenticating to Certificate Systems" would also be helpful, and bring the language in line with the CAB Forum requirements 1.5.1.

### LINKED ISSUE #299: Section 6.6.2
**Organization / Program:** FPKIMA <br>**Section:** 6.6.2 <br>**PDF Page:** 66<br>**PDF Line(s):** 1738<br>**Comment:** Additional requirest form CAB Forum NSR include Lockout account access to Certificate Systems after no more than five (5) failed access attempts, Implement automated mechanisms under the control of CA or Delegated Third Party Trusted Roles to process logged system activity and alert personnel, using notices provided to multiple destinations, of possible Critical Security Events; <br><br>**Suggested Change:** Add this to be compliant with trust stores

### LINKED ISSUE #302: Section 6.7
**Organization / Program:** FPKIMA <br>**Section:** 6.7 <br>**PDF Page:** 67<br>**PDF Line(s):** 1765-1767<br>**Comment:** Current wording will allow a network guard or a firewall or a filtering router (i.e., ACLs on a router).  A filtering router is not sufficient to protect the network access to CA equipment given the capabilities of modern and NextGen Firewalls, as well as modern day threats.  Even non-NextGen firewalls offer better options for security than just a router ACL. <br><br>**Suggested Change:** Change text to:

A network guard or firewall shall protect network access to CA equipment, and limit services allowed to and from the CA equipment to those required to perform CA functions.

### LINKED ISSUE #409: Section 6.1.1.2
**Organization / Program:** DoD <br>**Section:** 6.1.1.2 <br>**PDF Page:** 60<br>**PDF Line(s):** 1544-1545<br>**Comment:** Need to be consistent regarding terminology. Are RAs used?<br><br>**Suggested Change:** Add requirements for subscribers

### LINKED ISSUE #410: Section 6.1.1.3
**Organization / Program:** DoD <br>**Section:** 6.1.1.3 <br>**PDF Page:** 60<br>**PDF Line(s):** 1547-1549<br>**Comment:** What are the requirements for subscribers when generating keys? Also, how does a CA know if the key is known to be weak?<br><br>**Suggested Change:** State that CA keys shall not be archived

### LINKED ISSUE #411: Section 6.1.4
**Organization / Program:** DoD <br>**Section:** 6.1.4 <br>**PDF Page:** 61<br>**PDF Line(s):** 1566-1571<br>**Comment:** The entire intent of this PKI is to comply with browser, mobile OS, and other application requirements for including its Root CA key natively in their trust stores, why is it not listed as an acceptable method for Root CA certificate delivery?<br><br>**Suggested Change:** Add requirements for subscribers

### LINKED ISSUE #413: Section 6.1.5
**Organization / Program:** DoD <br>**Section:** 6.1.5 <br>**PDF Page:** 61-62<br>**PDF Line(s):** N/A<br>**Comment:** Since the profiles cover RSA and SHA-256, simplify the CP by deleting other material<br><br>**Suggested Change:** See comment

### LINKED ISSUE #414: Section 6.1.5, tables
**Organization / Program:** NSA <br>**Section:** 6.1.5, tables <br>**PDF Page:** 61<br>**PDF Line(s):** 1574-1578<br>**Comment:** Is there a plan to use DSA?  Strongly recommend deleting for Root and subordinate CA's.<br><br>**Suggested Change:** If there is no plan to use this, recommend deleting the rows in the tables.

### LINKED ISSUE #418: Section 6.2.1
**Organization / Program:** NSA <br>**Section:** 6.2.1 <br>**PDF Page:** 63<br>**PDF Line(s):** 1603<br>**Comment:** Is there a FIPS 140-2 level that subscriber modules need to be validated to?<br><br>**Suggested Change:** If so, list that validation level in this section.

### LINKED ISSUE #419: Section 6.2.3
**Organization / Program:** DoD <br>**Section:** 6.2.3 <br>**PDF Page:** 63<br>**PDF Line(s):** 1617-1619<br>**Comment:** Text in this section only addresses CA private keys. What about subscriber private keys? Note that Section 4.1.2 states that no keys will be escrowed.<br><br>**Suggested Change:** Add requirements for subscribers

### LINKED ISSUE #421: Section 6.2.4
**Organization / Program:** DoD <br>**Section:** 6.2.4 <br>**PDF Page:** 63<br>**PDF Line(s):** 1620-1628<br>**Comment:** Text in this section only addresses CA private keys. What about Subscriber private keys?<br><br>**Suggested Change:** Add requirements for subscribers

### LINKED ISSUE #422: Section 6.2.5, 6.2.6, 6.2.7, 6.2.8
**Organization / Program:** NSA <br>**Section:** 6.2.5, 6.2.6, 6.2.7, 6.2.8 <br>**PDF Page:** 64<br>**PDF Line(s):** 1629-1656<br>**Comment:**  If subscriber keys can be archived, transferred, stored, activated, deactivated, and/or destroyed, then state this in these sections.<br><br>**Suggested Change:** None

### LINKED ISSUE #423: Section 6.2.5
**Organization / Program:** DoD <br>**Section:** 6.2.5 <br>**PDF Page:** 64<br>**PDF Line(s):** 1630<br>**Comment:** Why would a CA private key be archived (even by the CA operator)?<br><br>**Suggested Change:** Add requirements for subscribers

### LINKED ISSUE #424: Section 6.2.5
**Organization / Program:** DoD <br>**Section:** 6.2.5 <br>**PDF Page:** 64<br>**PDF Line(s):** 1630<br>**Comment:** Text in this section only addresses CA private keys. What about Subscriber private keys?<br><br>**Suggested Change:** Replace with text that is consistent with section 5.5.1

### LINKED ISSUE #425: Section 6.2.6
**Organization / Program:** DoD <br>**Section:** 6.2.6 <br>**PDF Page:** 64<br>**PDF Line(s):** 1631<br>**Comment:** Text in this section only addresses CA private keys. What about Subscriber private keys?<br><br>**Suggested Change:** Add requirements for subscribers

### LINKED ISSUE #426: Section 6.2.6
**Organization / Program:** DoD <br>**Section:** 6.2.6 <br>**PDF Page:** 64<br>**PDF Line(s):** 1634<br>**Comment:** Text references section 6.2.4.1, but there is no section 6.2.4.1 in this document<br><br>**Suggested Change:** Add requirements for subscribers

### LINKED ISSUE #427: Section 6.2.7
**Organization / Program:** DoD <br>**Section:** 6.2.7 <br>**PDF Page:** 64<br>**PDF Line(s):** 1641<br>**Comment:** Text in this section only addresses CA private keys. What about Subscriber private keys?<br><br>**Suggested Change:** Define "offline" to mean a CA that is disconnected from all other networks and can be either powered on generally or only powered on when needed and remove the term "air-gapped"

### LINKED ISSUE #428: Section 6.2.8
**Organization / Program:** DoD <br>**Section:** 6.2.8 <br>**PDF Page:** 64<br>**PDF Line(s):** 1646<br>**Comment:** Text in this section only addresses CA private keys. What about Subscriber private keys?<br><br>**Suggested Change:** Specify what, if any, requirements CAs have for time-stamping information

### LINKED ISSUE #429: Section 6.4.2
**Organization / Program:** DoD <br>**Section:** 6.4.2 <br>**PDF Page:** 65<br>**PDF Line(s):** 1674<br>**Comment:** Text in this section only addresses CA activation data. What about subscriber activation data?<br><br>**Suggested Change:** Consider allowing the third option. If not, need to delete the word "or" from the end of the second option

### LINKED ISSUE #431: Section 6.6.3
**Organization / Program:** NSA <br>**Section:** 6.6.3 <br>**PDF Page:** 67<br>**PDF Line(s):** 1747<br>**Comment:** Penetration testing every 90 days seems excessive.  And I've actively discouraged the permission to perform penetration testing on a live Root CA.<br><br>**Suggested Change:** None

### LINKED ISSUE #489: Definition: Certificate System
**Organization / Program**: (n/a)
**Section**: Definitions

Current definition has Certificate System defined as: 

> **Certificate System**: A system used by a CA to process, approve issuance of, or store certificates or certificate status information, including the database, database server, and storage.

Updated definition: 
[Network and Certificate System Security Requirements](https://cabforum.org/wp-content/uploads/CABForum_Network_Security_Controls_v.1.1.pdf) 

> **Certificate Systems**: The system used by a CA in providing identity verification, registration and enrollment, certificate approval, issuance, validity status, support, and other PKI related services.

### PR DESCRIPTION
**Pull request by lachellel**
- 55 comments reviewed
- 45 comments incorporated as accepted or no change

closes #302 closes #259 closes #409 closes #410 closes #260 closes #296 closes #411 closes #413 closes #414 closes #418 closes #261 closes #419 closes #262 closes #421 closes #423 closes #424 closes #422 closes #425 closes #426 closes #427 closes #428 closes #297 closes #244 closes #429 closes #298 closes #263 closes #245 closes #299 closes #431 partial #469 closes #489

### CODE REVIEW

File: certificate-policy.md [hunk 1]
```diff
@@ -24,21 +24,21 @@ The terms and provisions of this certificate policy shall be interpreted under a
...
 ### 1.1.2.1.2 Subordinate CA Certificates
 
-### 1.1.2.2 End-Entity Certificates
+### 1.1.2.2 Subscriber Certificates
 
-### 1.1.2.2.1 Domain Validation TLS Server Authentication Certificates
+### 1.1.2.2.1 Domain Validation SSL Server Authentication Certificates
 
-### 1.1.2.2.2 Organization Validation TLS Server Authentication Certificates
+### 1.1.2.2.2 Organization Validation SSL Server Authentication Certificates
```
Comment (konklone) [inline certificate-policy.md]: Are you changing TLS -> SSL to hurt me? ;)

I didn't see a request for this change in any of the issues that this PR links to as fixing. Was there a comment to this effect, with a good rationale?

File: certificate-policy.md [hunk 2]
```diff
@@ -1153,271 +1153,267 @@ Prior to CA termination, the CA shall provide archived data to an archive facili
...
+| :---  | :------ |  
+| Digest algorithm | SHA-256, SHA-384 or SHA-512  |  
+| Minimum RSA modulus size (bits) | 2048 |  
+| ECC curve | NIST P-256, P-384, or P-521 |  
 
-\*\*\* L and N (the bit lengths of modulus p and divisor q, respectively) are described in the Digital Signature Standard, FIPS 186-4 (http://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf).
 
 ### 6.1.6 Public key parameters generation and quality checking
-RSA: The CA shall confirm that the value of the public exponent _e_ is an odd positive integer such that:  
+For RSA modulus, the CA shall confirm that the value of the public exponent _e_ is an odd positive integer such that:
```
Comment (konklone) [inline certificate-policy.md]: > For RSA modulus,

This looks wrong grammatically. Should this be "When using an RSA modulus", or "For RSA moduluses", or...?

File: certificate-policy.md [hunk 3]
```diff
@@ -1153,271 +1153,267 @@ Prior to CA termination, the CA shall provide archived data to an archive facili
...
-- Configurations	shall be reviewed	on	at	least	a	weekly	basis	to	determine	whether	any	changes	violated	the	CA’s	security	policies.
-- There shall be a mechanism for detecting unauthorized modification to the software or configuration.
-Configurations	shall be reviewed	on	at	least	a	weekly	basis	to	determine	whether	any	changes	violated	the	CA’s	security	policies.
-- All system and trusted role accounts shall be reviewed at least every ninety (90) days. Any account that is no longer in use or necessary for operations shall be deactivated.
-- A process shall be implemented that disables physical and logical access to a Certificate System by either a privileged user or a trusted role within 24 hours upon termination of the individual's employment or contracting relationship with the CA.
-- All authentication credentials for any account or trusted role on a Certificate System shall be changed whenever authorization to access the account is changed or revoked.
+- Configurations, modifications, and upgrades shall be documented and controlled
+- Configurations shall be reviewed on at least a weekly	basis to determine whether any changes violated	the	CA’s security policies.
```
Comment (konklone) [inline certificate-policy.md]: Nit: unnecessary whitespace in the middle of the sentence.

**Review by konklone — COMMENTED** [review]
I left one question about TLS->SSL, and the other two are grammar/nits. Would love some clarifications there, but otherwise :+1:.

File: certificate-policy.md (hunk 1 above)
Comment (lachellel) [reply certificate-policy.md]: I'm changing back to TLS!  I was trying to match communications I see that are using different descriptions.  We agreed to TLS and will stick to it.

File: certificate-policy.md (hunk 2 above)
Comment (lachellel) [reply certificate-policy.md]: moduli is plural of modulus... :+1:
